client/state/partner-portal/licenses/selectors.ts

```diff
@@ -28,3 +29,26 @@ export function getLicenseCounts( state: PartnerPortalStore ): LicenseCounts {
 export function hasFetchedLicenseCounts( state: PartnerPortalStore ): boolean {
 	return state.partnerPortal.licenses.hasFetchedLicenseCounts;
 }
+
+// Returns the product IDs of the assigned plan(bundle) & all the products to a particular site in an array.
+export function getAssignedPlanAndProductIDsForSite(
+	state: AppState,
+	siteId: number
+): Array< number > {
+	let planAndProductIDs: number[] = [];
+	const currentSite = state?.sites?.items?.[ siteId ];
+	// Get the assigned plan(bundle) to a site.
+	const plan = currentSite?.plan.product_id;
+	if ( plan ) {
+		planAndProductIDs.push( plan );
+	}
+	// Get all assigned products to a site.
+	const products = currentSite?.products.map( ( product: { product_id: string } ) =>
+		// parseInt since the product IDs are strings of numbers.
+		parseInt( product.product_id )
+	);
+	if ( products?.length ) {
+		planAndProductIDs = [ ...planAndProductIDs, ...products ];
+	}
+	return planAndProductIDs;
```

Comment:
I think we can get rid of the `products` variable and populate the data in the `planAndProductIDs` array directly.

```suggestion
	const planAndProductIDs: number[] = [];
	const currentSite = state?.sites?.items?.[ siteId ];
	// Get the assigned plan(bundle) to a site.
	const plan = currentSite?.plan.product_id;
	if ( plan ) {
		planAndProductIDs.push( plan );
	}
	// Get all assigned products to a site.
	currentSite?.products.forEach( ( product: { product_id: string } ) =>
		planAndProductIDs.push( parseInt( product.product_id ) )
	);
	return planAndProductIDs;
```